lib/AST/Type.cpp

```diff
@@ -4471,6 +4471,17 @@ Optional<VectorSpace> TypeBase::getAutoDiffAssociatedTangentSpace(
   return cache(None);
 }
 
+Type getTangentType(Type type) {
```

Comment:
This should be defined as a closure inside `AnyFunctionType::getAutoDiffAssociatedFunctionType`.